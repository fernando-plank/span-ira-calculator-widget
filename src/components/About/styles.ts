import styled from 'styled-components'

export const Wrapper = styled.div`
  width: 100%;
  display: flex;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    background: url(${({ theme }) => theme.banner.mobile}) no-repeat;
    padding: 347px 30px 20% 30px;
    text-align: left;
    flex-direction: column;
    background-size: 100%;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
    background: url(${({ theme }) => theme.banner.desktop}) no-repeat;
    flex-direction: row-reverse;
    padding: 267px 64px 20% 64px;
    background-size: 100%;
  }
`

export const Title = styled.h1`
  font-weight: 600;
  color: ${props => props.theme.colors.white};

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    width: 55%;
    font-size: 70px;
    line-height: 70px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
    width: 90%;
    font-size: 136px;
    line-height: 136px;
  }
`

export const DescriptionWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

export const Description = styled.p`
  color: ${props => props.theme.colors.white};
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    width: 100%;
    margin-top: 24px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
    width: 50%;
    margin-top: 0px;
  }
`

export const ReadBlogButton = styled.button`
  border: 2px solid ${({ theme }) => theme.colors.white};
  border-radius: 20px;
  background: none;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
  color: ${({ theme }) => theme.colors.white};
  height: 40px;
  margin-top: 40px;
  margin-bottom: 104px;
  cursor: pointer;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    width: 50%;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
    width: 30%;
  }
`
