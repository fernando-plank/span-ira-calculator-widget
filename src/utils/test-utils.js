var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
/* eslint-disable import/export */
import React from 'react';
import theme from '@styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
jest.mock('../components/Icon', () => {
    return jest.fn(() => React.createElement("div", null, "mocked icon"));
});
const customRender = (ui, _a = {}) => {
    var renderOptions = __rest(_a, []);
    return render(React.createElement(ThemeProvider, { theme: theme }, ui), renderOptions);
};
export * from '@testing-library/react';
export { customRender as render };
