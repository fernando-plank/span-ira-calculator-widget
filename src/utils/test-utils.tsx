/* eslint-disable import/export */
import { ReactElement } from 'react'
import { RenderOptions, render } from '@testing-library/react'

import theme from '@styles/theme'
import { ThemeProvider } from 'styled-components'

jest.mock('../components/Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

const customRender = (ui: ReactElement, { ...options }: RenderOptions = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options)

export * from '@testing-library/react'
export { customRender as render }
