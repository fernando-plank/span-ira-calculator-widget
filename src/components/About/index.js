import React from 'react';
import * as S from './styles';
const About = () => (React.createElement(S.Wrapper, null,
    React.createElement(S.Title, { "data-testid": "about-title" }, "SPAN + IRA"),
    React.createElement(S.DescriptionWrapper, null,
        React.createElement(S.Description, null, "Get the most up-to-date facts and information about how the IRA affects you and your home by reading SPAN\u00AE & the Inflation Reduction Act, a new addition to our blog written by SPAN CEO and Founder Arch Rao."),
        React.createElement(S.ReadBlogButton, { onClick: () => window.open('https://www.span.io/blog/span-and-the-inflation-reduction-act', '_self') }, "Read blog post"))));
export default About;
