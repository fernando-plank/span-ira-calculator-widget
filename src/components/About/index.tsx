import React from 'react'
import * as S from './styles'

const About = () => (
  <S.Wrapper>
    <S.Title data-testid="about-title">SPAN + IRA</S.Title>
    <S.DescriptionWrapper>
      <S.Description>
        Get the most up-to-date facts and information about how the IRA affects
        you and your home by reading SPANÂ® & the Inflation Reduction Act, a new
        addition to our blog written by SPAN CEO and Founder Arch Rao.
      </S.Description>
      <S.ReadBlogButton
        onClick={() =>
          window.open(
            'https://www.span.io/blog/span-and-the-inflation-reduction-act',
            '_self'
          )
        }
        data-testid="read-blog-button"
      >
        Read blog post
      </S.ReadBlogButton>
    </S.DescriptionWrapper>
  </S.Wrapper>
)
export default About
