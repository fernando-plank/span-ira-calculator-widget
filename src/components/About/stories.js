import React from 'react';
import About from '.';
export default {
    title: 'About',
    component: About
};
export const Default = () => React.createElement(About, null);
