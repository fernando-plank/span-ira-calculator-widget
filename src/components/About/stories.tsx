import React from 'react'
import { Story, Meta } from '@storybook/react'

import About from '.'

export default {
  title: 'About',
  component: About
} as Meta

export const Default: Story = () => <About />
