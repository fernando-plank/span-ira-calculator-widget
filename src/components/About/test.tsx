import React from 'react'
import { render } from '../../utils/test-utils'

import { screen } from '@testing-library/react'

import About from '.'

describe('<About />', () => {
  it('should render the title', () => {
    const { container } = render(<About />)

    expect(screen.getByTestId('about-title')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
