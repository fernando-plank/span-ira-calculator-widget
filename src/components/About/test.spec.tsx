import { render } from '../../utils/test-utils'
import { fireEvent, screen } from '@testing-library/react'

import About from '.'

describe('<About />', () => {
  it('should render the title', () => {
    const { container } = render(<About />)

    expect(screen.getByTestId('about-title')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should call open on click button', () => {
    const { container } = render(<About />)
    const openSpy = jest.spyOn(window, 'open')

    const readBlogButton = screen.getByTestId('read-blog-button')
    expect(readBlogButton).toBeInTheDocument()

    fireEvent.click(readBlogButton)

    expect(container.firstChild).toMatchSnapshot()
    expect(openSpy).toBeCalled()
  })
})
