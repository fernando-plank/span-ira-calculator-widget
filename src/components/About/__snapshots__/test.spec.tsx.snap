// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should render the title 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  font-weight: 600;
  color: #FAFAFA;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  color: #FAFAFA;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
}

.c4 {
  border: 2px solid #FAFAFA;
  border-radius: 20px;
  background: none;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
  color: #FAFAFA;
  height: 40px;
  margin-top: 40px;
  margin-bottom: 104px;
  cursor: pointer;
}

@media (min-width:320px) {
  .c0 {
    background: url() no-repeat;
    padding: 347px 30px 20% 30px;
    text-align: left;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    background-size: 100%;
  }
}

@media (min-width:768px) {
  .c0 {
    background: url() no-repeat;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    padding: 267px 64px 20% 64px;
    background-size: 100%;
  }
}

@media (min-width:320px) {
  .c1 {
    width: 55%;
    font-size: 70px;
    line-height: 70px;
  }
}

@media (min-width:768px) {
  .c1 {
    width: 90%;
    font-size: 136px;
    line-height: 136px;
  }
}

@media (min-width:320px) {
  .c3 {
    width: 100%;
    margin-top: 24px;
  }
}

@media (min-width:768px) {
  .c3 {
    width: 50%;
    margin-top: 0px;
  }
}

@media (min-width:320px) {
  .c4 {
    width: 50%;
  }
}

@media (min-width:768px) {
  .c4 {
    width: 30%;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
    data-testid="about-title"
  >
    SPAN + IRA
  </h1>
  <div
    class="c2"
  >
    <p
      class="c3"
    >
      Get the most up-to-date facts and information about how the IRA affects you and your home by reading SPANÂ® & the Inflation Reduction Act, a new addition to our blog written by SPAN CEO and Founder Arch Rao.
    </p>
    <button
      class="c4"
    >
      Read blog post
    </button>
  </div>
</div>
`;
